blocks/model.py

```diff
@@ -161,15 +162,18 @@ def __init__(self, outputs):
             raise ValueError("top bricks with the same name:"
                              " {}".format(', '.join(repeated_names)))
 
-        brick_param_names = {
-            v: k for k, v in Selector(self.top_bricks).get_params().items()}
-        self.params = []
-        for param in VariableFilter(roles=[PARAMETER])(self.shared_variables):
-            if param in brick_param_names:
-                self.params.append((brick_param_names[param], param))
+        brick_parameter_names = {
+            v: k
+            for k, v in Selector(self.top_bricks).get_parameters().items()}
+        self.parameters = []
```

Comment:
@rizar One last thing. Is there any reason why it would be useful to build ComputationGraph.parameters as a list rather than a dictionary? We could still use ComputationGraph.parameters.values() when we only care about a list. This would simplify the common interface. Otherwise, I find names of get_parameter_dict and get_parameter_values to be quite confusing.